CONTRIBUTING.md

```diff
@@ -125,10 +125,13 @@ versioning guidelines:
 
 # Runtime guarding
 
-Some high risk changes in Envoy are deemed worthy of runtime guarding. Instead of just replacing
+Some changes in Envoy are deemed worthy of runtime guarding. Instead of just replacing
 old code with new code, both code paths are supported for between one Envoy release (if it is
 guarded due to performance concerns) and a full deprecation cycle (if it is a high risk behavioral
-change).
+change). Generally as a community we try to guard both high risk changes (major
+refactors such as replacing Envoy's buffer implementation) and most user-visible
+non-config-guarded changes to protocol processing (for example additions or changes to HTTP headers or
+how HTTP is serialized out) for non-alpha features.
```

Comment:
A couple of other thoughts:
1) Should we have some blurb about maintainer discretion / consulting maintainers as needed?
2) Should we update the PR template with a "feature flag / runtime" field to get people thinking about this? It wouldn't be filled out that often but it would require people to add "N/A" explicitly? WDYT?